File: code-breaker/typescript/tests/guessBuilder.ts
import { createGuess, Peg, type Guess } from '../src/codeBreaker.js';

export class GuessBuilder {
  private _pegs: readonly [Peg, Peg, Peg, Peg] = [Peg.One, Peg.Two, Peg.Three, Peg.Four];

  with(a: Peg, b: Peg, c: Peg, d: Peg): this {
    this._pegs = [a, b, c, d];
    return this;
  }

  build(): Guess {
    return createGuess(this._pegs);
  }
}

File: code-breaker/python/tests/guess_builder.py
from code_breaker import Guess, Peg


class GuessBuilder:
    def __init__(self) -> None:
        self._pegs = (Peg.ONE, Peg.TWO, Peg.THREE, Peg.FOUR)

    def with_pegs(self, a: Peg, b: Peg, c: Peg, d: Peg) -> "GuessBuilder":
        self._pegs = (a, b, c, d)
        return self

    def build(self) -> Guess:
        return Guess(pegs=self._pegs)
